tests/foreman/ui/test_rhcloud_insights.py

```diff
@@ -215,3 +217,221 @@ def test_host_sorting_based_on_recommendation_count():
 
     :CaseAutomation: ManualOnly
     """
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.tier2
+def test_host_details_page(
+    rhel8_insights_vm,
+    fixable_rhel8_vm,
+    organization_ak_setup,
+    set_rh_cloud_token,
+    rhcloud_sat_host,
+):
+    """Test host details page for host having insights recommendations.
+
+    :id: e079ed10-c9f5-4331-9cb3-70b224b1a584
+
+    :Steps:
+        1. Prepare misconfigured machine and upload its data to Insights
+        2. Add Cloud API key in Satellite
+        3. Sync insights recommendations.
+        4. Sync RH Cloud inventory status.
+        5. Go to Hosts -> All Hosts
+        6. Assert there is "Recommendations" column containing insights recommendation count.
+        7. Assert that host properties shows reporting inventory upload status.
+        8. Click on "Recommendations" tab.
+
+    :expectedresults:
+        1. There's Insights column with number of recommendations.
+        2. Inventory upload status is present in host properties table.
+        3. Clicking on "Recommendations" tab takes user to Insights page with insights
+            recommendations selected for that host.
+
+    :BZ: 1974578
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+    # Sync inventory status
+    inventory_sync = rhcloud_sat_host.api.Organization(id=org.id).rh_cloud_inventory_sync()
+    wait_for(
+        lambda: rhcloud_sat_host.api.ForemanTask()
+        .search(query={'search': f'id = {inventory_sync["task"]["id"]}'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
+    with Session(hostname=rhcloud_sat_host.hostname) as session:
+        session.organization.select(org_name=org.name)
+        session.location.select(loc_name=DEFAULT_LOC)
+        # Sync insights recommendations
+        timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M')
+        session.cloudinsights.sync_hits()
+        wait_for(
+            lambda: rhcloud_sat_host.api.ForemanTask()
+            .search(query={'search': f'Insights full sync and started_at >= "{timestamp}"'})[0]
+            .result
+            == 'success',
+            timeout=400,
+            delay=15,
+            silent_failure=True,
+            handle_exception=True,
+        )
+        result = session.host.search(rhel8_insights_vm.hostname)[0]
+        assert result['Name'] == rhel8_insights_vm.hostname
+        assert int(result['Recommendations']) > 0
+        values = session.host.get_details(rhel8_insights_vm.hostname)
+        # Note: Reading host properties adds 'clear' to original value.
```

Comment:
yes, there's clear button and it's text is being read by `get_details`. We will have to fix this issue in airgun.

Reply:
When the UI changes, or you change airgun, the test will fail. So you could  assert the expected_text, with out "clear" is in result_text so that it keeps working.  I worked around this issue when "clear" was added in errata test, see https://github.com/SatelliteQE/robottelo/blob/ecbf6f7fe7f348597f8f619ee404106cd8cf0788/tests/foreman/ui/test_errata.py#L921

But no need to block the test for that.

Reply:
Thanks for the suggestion, but I can't use it for all of assertions ex. https://github.com/SatelliteQE/robottelo/pull/9161/files#diff-bf79c024a088d41de18007b5dd0ffcf658d6626305a5796548de6499ae673ee9R432
It can be `Reporting` and `Not Reporting` so I can't use `assert "Reporting" in result`.
If something changes in future then I'll update this test then.